tests/unit/test_server.py

```diff
@@ -0,0 +1,279 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Jenkins-k8s-agent server module tests."""
+
+# Need access to protected functions for testing
+# pylint:disable=protected-access
+
+import secrets
+import typing
+import unittest.mock
+
+import ops
+import ops.testing
+import pytest
+import requests
+
+import server
+import state
+from charm import JenkinsAgentCharm
+
+
+def test_credentials_from_jenkins_slave_interface_dict_partial(harness: ops.testing.Harness):
+    """
+    arrange: given a slave relation databag containing partial url data.
+    act: when from_jenkins_slave_interface_dict is called.
+    assert: None is returned.
+    """
+    relation_id = harness.add_relation(state.SLAVE_RELATION, "jenkins")
+    harness.add_relation_unit(relation_id=relation_id, remote_unit_name="jenkins/0")
```

Comment:
What is the reason for needing the relation here? Generally speaking the modules that aren't the charm should only need the container at most

Reply:
I wanted to use the relation data set by the harness testing framework since the initialization of `RelationDataContent` depends on `Relation`, `Entity`(Unit/Application) and `ModelBackend`. In retrospect, I could just mock it. Thank you for the comment!

Reply:
This test is removed since server no longer requires charm. Thank you!